bin/migrate-oats-data/noi/set_hide_from_portal_on_noi.py

```diff
@@ -0,0 +1,48 @@
+from common import setup_and_get_logger
+from db import inject_conn_pool
+from psycopg2.extras import RealDictCursor
+
+
+etl_name = "set_hide_from_portal_on_notice_of_intent"
+logger = setup_and_get_logger(etl_name)
+
+
+@inject_conn_pool
+def set_hide_from_portal_on_notice_of_intent(conn=None):
+    """
+    This function is responsible for setting hide_from_portal colum on notice_of_intent table where oats.who_create is not PROXY user
+
+    Args:
+    conn (psycopg2.extensions.connection): PostgreSQL database connection. Provided by the decorator.
+    """
+    logger.info(f"Start {etl_name}")
+
+    try:
+        with conn.cursor(cursor_factory=RealDictCursor) as cursor:
+            count_query = f"""
+                    SELECT count(*) FROM oats.oats_alr_applications oaa 
+                    JOIN alcs.notice_of_intent a ON a.file_number = oaa.alr_application_id::TEXT 
+                    WHERE oaa.who_created <> 'PROXY_OATS_APPLICANT';
+                """
+            cursor.execute(count_query)
+            count_total = dict(cursor.fetchone())["count"]
+            logger.info(f"Total application to set hide_from_portal: {count_total}")
+
+            update_query = f"""
+                                UPDATE alcs.notice_of_intent AS noi
+                                SET hide_from_portal = TRUE
+                                FROM oats.oats_alr_applications AS oaa
+                                WHERE noi.file_number = oaa.alr_application_id::TEXT
+                                    AND oaa.who_created <> 'PROXY_OATS_APPLICANT';
```

Comment:
Is there a reason you do a loop here for each noi? Wouldn't it be more efficient to execute this once without the file_number in the where clause to set all of them?

Reply:
this is a join of ALCS with OATS data. Since we need to set who created  in ALCS only for 'PROXY_OATS_APPLICANT' that is stored in OATS